satip/goes_download_manager.py

```diff
@@ -0,0 +1,90 @@
+"""
+Script for downloading GOES data.
+"""
+
+import datetime
+import logging
+import os
+
+from goes2go import GOES
+
+
+class GOESDownloadManager:
+    """
+    Manager class for downloading GOES data.
+    """
+    def __init__(self, data_dir, log_directory=None):
+        """
+        Initialize the GOESDownloadManager.
+
+        Args:
+            data_dir (str): Directory to save downloaded GOES data.
+            log_directory (str, optional): Directory to save logs.
+            If None, logging is printed to STDOUT.
+        """
+        self.data_dir = data_dir
+        self.ensure_directory_exists(self.data_dir)
+
+        if log_directory:
+            self.ensure_directory_exists(log_directory)
+            logging.basicConfig(
+                filename=os.path.join(log_directory, 'goes_download.log'),
+                level=logging.INFO)
+        else:
+            logging.basicConfig(level=logging.INFO)
+
+        logging.info(f"GOESDownloadManager initialized. Data will be saved to: {data_dir}")
+
+    @staticmethod
+    def ensure_directory_exists(directory):
+        """Ensures the specified directory exists, creating it if necessary."""
+        if not os.path.exists(directory):
+            try:
+                os.makedirs(directory)
+                logging.info(f"Created directory: {directory}")
+            except Exception as e:
+                logging.error(f"Error creating directory {directory}: {e}")
+                raise
+    def download_goes_data(self, start_time, end_time, product='ABI-L1b-RadC',
+                           domain='F', satellite=16):
+        """
+        Download GOES data for a specified time range and product.
+
+        Args:
+            start_time (datetime): Start of the download period.
+            end_time (datetime): End of the download period.
+            product (str): GOES product identifier. Default is 'ABI-L1b-RadC'.
+            domain (str): Domain for the product. Default is 'F' (Full Disk).
+            satellite (int): GOES satellite number. Default is 16.
+        """
+        G = GOES(satellite=satellite, product=product, domain=domain)
+        current_time = start_time
+        while current_time <= end_time:
+            try:
+                # Download the data
+                ds = G.nearesttime(current_time)
+
+                # Format the date string for filename
+                date_string = current_time.strftime("%Y-%m-%d_%H-%M-%S")
```

Comment:
Rather than having the date string be from the `current_time`, it probably makes more sense to have it from the satellite image itself. The dataset should contain the acquisition time of the image, and that would be more accurate than this one. That would also make it easier to check if the data has already been downloaded.